### PR TITLE
Convert Pilot mesh ip attributes to BYTES

### PR DESCRIPTION
**Pull request by qiwzhang**

### CODE REVIEW

**Comment by qiwzhang** [comment]
https://github.com/istio/proxy/issues/483

File: src/envoy/mixer/mixer_control.cc [hunk 1]
```diff
@@ -382,6 +388,32 @@ void MixerControl::BuildTcpReport(
...
 }
 
+// Mesh attributes from Pilot are all string type.
+// The attributes with ".ip" suffix will be treated
+// as ipv4 and use BYTES attribute type.
+void MixerControl::SetMeshAttribute(const std::string& name,
+                                    const std::string& value,
+                                    Attributes* attr) const {
+  // Check with ".ip" suffix,
+  if (name.length() < kIPSuffix.length() ||
```
Comment (mangchiandjjoe) [inline src/envoy/mixer/mixer_control.cc]: "name" can be ".ip"? otherwise '<=' seems to be better,
Does name.compare need to be case insensitive?

Comment (qiwzhang) [reply src/envoy/mixer/mixer_control.cc]: Thanks.  ".ip" is not allowed,  code is changed.
All istio attribute names are lower cases.  
https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html

**Comment by costinm** [comment]
Wait - some context ? This seems like an API change - is this based on a bug, and was it approved (we're locking down the release).

**Comment by qiwzhang** [comment]
@costinm This hack is needed for 0.2. Here is the detail. https://github.com/istio/proxy/issues/483

File: src/envoy/mixer/mixer_control.cc [hunk 2]
```diff
@@ -382,6 +388,32 @@ void MixerControl::BuildTcpReport(
...
+                                    Attributes* attr) const {
+  // Check with ".ip" suffix,
+  if (name.length() <= kIPSuffix.length() ||
+      name.compare(name.length() - kIPSuffix.length(), kIPSuffix.length(),
+                   kIPSuffix) != 0) {
+    attr->attributes[name] = Attributes::StringValue(value);
+    return;
+  }
+
+  in_addr ipv4_bytes;
```
Comment (costinm) [inline src/envoy/mixer/mixer_control.cc]: Please add a block for ipv6.

The sidecar will run on raw VMs, some on prem using ipv6 - we can't add new code that doesn't support ipv6.

Comment (qiwzhang) [reply src/envoy/mixer/mixer_control.cc]: https://github.com/istio/proxy/pull/503

**Pull request merged**